org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -10484,3 +10510,682 @@ class DiagnosticWorkspaceCapabilities {
 		this.refreshSupport = refreshSupport
 	}
 }
+
+/**
+ * A notebook document.
+ * <p>
+ * Since 3.17.0
+ */
+@JsonRpcData
+class NotebookDocument {
+	/**
+	 * The notebook document's URI.
+	 */
+	@NonNull
+	String uri
+
+	/**
+	 * The type of the notebook.
+	 */
+	@NonNull
+	String notebookType
+
+	/**
+	 * The version number of this document (it will increase after each
+	 * change, including undo/redo).
+	 */
+	int version
+
+	/**
+	 * Additional metadata stored with the notebook
+	 * document.
+	 */
+	@JsonAdapter(JsonElementTypeAdapter.Factory)
+	Object metadata
+
+	/**
+	 * The cells of a notebook.
+	 */
+	@NonNull
+	List<NotebookCell> cells
+
+	new() {
+		this.cells = new ArrayList
+	}
+
+	new(@NonNull String uri, @NonNull String notebookType, int version, @NonNull List<NotebookCell> cells) {
+		this.uri = Preconditions.checkNotNull(uri, 'uri')
+		this.notebookType = Preconditions.checkNotNull(notebookType, 'notebookType')
+		this.version = version
+		this.cells = Preconditions.checkNotNull(cells, 'cells')
+	}
+}
+
+/**
+ * A notebook cell.
+ * <p>
+ * A cell's document URI must be unique across ALL notebook
+ * cells and can therefore be used to uniquely identify a
+ * notebook cell or the cell's text document.
+ * <p>
+ * Since 3.17.0
+ */
+@JsonRpcData
+class NotebookCell {
+	/**
+	 * The cell's kind
+	 * <p>
+	 * See {@link NotebookCellKind} for allowed values.
+	 */
+	int kind
+
+	/**
+	 * The URI of the cell's text document
+	 * content.
+	 */
+	@NonNull
+	String document
+
+	/**
+	 * Additional metadata stored with the cell.
+	 */
+	@JsonAdapter(JsonElementTypeAdapter.Factory)
+	Object metadata
+
+	/**
+	 * Additional execution summary information
+	 * if supported by the client.
+	 */
+	ExecutionSummary executionSummary
+
+	new() {
+	}
+
+	new(int kind, @NonNull String document) {
+		this.kind = kind
+		this.document = Preconditions.checkNotNull(document, 'document')
+	}
+}
+
+/**
+ * A notebook cell kind.
+ * <p>
+ * Since 3.17.0
+ */
+final class NotebookCellKind {
```

Comment:
Good points! Thanks for the clarification.

FWIW, the current approach will not fail on an unknown enumeration value - `null` is returned in this case (see `EnumTypeAdapter`), but it cannot preserve the value on round trips (the latter being a recommendation, not a requirement of the spec).

IMHO, it would be a more clear workflow to make it an enum for consistency sake now and discuss in a separate issue if a new pattern for enums needs to be established.